### ISSUE TITLE
MB-MelGAN consumes a lot of memory while starting evaluation

### ISSUE THREAD

**Issue by OscarVanL** [TensorSpeech/TensorFlowTTS#359]
(Issue opened after https://github.com/TensorSpeech/TensorFlowTTS/issues/354 as this appears to be a separate issue)

I am fine-tuning the `multiband_melgan.v1_24k` Universal Vocoder with command:

```
CUDA_VISIBLE_DEVICES=0 python examples/multiband_melgan/train_multiband_melgan.py \
  --train-dir ./dump_LibriTTSFormatted/train/ \
  --dev-dir ./dump_LibriTTSFormatted/valid/ \
  --outdir ./outdir/MBMELGAN/MBMelgan-Tune-Experiment1 \
  --config ./models/multiband_melgan.v1_24k.yaml \
  --use-norm 1 \
  --pretrained ./models/libritts_24k.h5
```

After 5000 steps it begins evaluation...

![image](https://user-images.githubusercontent.com/7999692/98443760-6bad3a80-2105-11eb-97fc-68a68000bdaa.png)

Sometimes it gets killed while filling the shuffle buffer. Sometimes it fills the buffer, but gets stuck in evaluation for some minutes before then getting killed.

I put training into the background and entered `top` as I reached the Shuffle buffer filled stage. I noticed using SSH was extremely laggy and slow. (Probably because the system memory is full). 
It says python is using almost 35GB of Virtual memory, ~15GB of RAM (the remaining RAM is full). 
![image](https://user-images.githubusercontent.com/7999692/98443885-3d7c2a80-2106-11eb-9773-f6fe5bc702af.png)

**Comment by OscarVanL** [comment]
As for debugging in headless SSH, should I try the [TensorFlow Profiler](https://www.tensorflow.org/guide/profiler#:~:text=Profiling%20helps%20you%20understand%20the,make%20the%20model%20execute%20faster.&text=Note%3A%20You%20cannot%20use%20the,data.)? It seems to show a lot of memory usage details.

**Comment by machineko** [comment]
https://www.jetbrains.com/help/pycharm/remote-debugging-with-product.html
Here is small tutorial for easier debugging using pycharm on a remote machine :)

**Comment by dathudeptrai** [comment]
can you check https://github.com/TensorSpeech/TensorFlowTTS/blob/master/examples/multiband_melgan/conf/multiband_melgan.v1.yaml#L74. Maybe because ur ram is not enough to catch all dataset ? @OscarVanL

**Comment by OscarVanL** [comment]
I think that solved the problem 👍 

When you have `allow_cache` enabled, does it cache the entire dataset, or just the train/eval depending on whether it is training or evaluating?

If it is the former, I noticed `dump_LibriTTSFormatted` (with train and valid data) is 17.17GB, so if it caches the entire data it is too large for my 16GB memory :) I assume this is the problem!

**Comment by OscarVanL** [comment]
> https://www.jetbrains.com/help/pycharm/remote-debugging-with-product.html
> Here is small tutorial for easier debugging using pycharm on a remote machine :)

Wow, this is a really nice method, I will try this out in the future :D
